include/lbann/utils/dim_helpers.hpp

```diff
@@ -99,6 +99,12 @@ auto get_packed_strides(std::vector<T> const& dims)
   return get_strides(dims.size(), dims.data(), T(1));
 }
 
+template <typename To, typename From>
+auto vector_cast(std::vector<From> const& from)
+{
+  return std::vector<To>{from.cbegin(), from.cend()};
+}
+
```

Comment:
Flagging this for future work: It would be useful if we had a narrow-safe version of this to use in debug builds.